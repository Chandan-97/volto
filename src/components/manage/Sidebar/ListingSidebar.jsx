import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { intlShape } from 'react-intl';

export default class ListingSidebar extends Component {
  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    data: PropTypes.objectOf(PropTypes.any).isRequired,
    tile: PropTypes.string.isRequired,
    intl: intlShape.isRequired,
  };

  /**
   * Constructor
   * @method constructor
   * @param {object} props Properties
   * @returns {undefined}
   */
  constructor(props) {
    super(props);
    this.state = {
      query: props.data.query || '',
    };
  }

  /**
   * Change handler
   * @method onChange
   * @param {object} e Change event
   * @returns {undefined}
   */
  onChange = e => {
    const value = e.currentTarget.value;
    this.setState({
      query: value,
    });
  };

  /**
   * Submit handler
   * @method onSubmit
   * @param {object} e Submit event
   * @returns {undefined}
   */
  onSubmit = e => {
    e.preventDefault();
    const { query } = this.state;
    this.props.onChangeTile(this.props.tile, {
      ...this.props.data,
      query,
    });
  };

  render() {
    return (
      <div className="listing-sidebar">
        <form className="query-form" onSubmit={this.onSubmit}>
          <input
            name="query"
            className="query"
            placeholder="querystring"
            onChange={this.onChange}
            value={this.state.query}
          />
        </form>
      </div>
    );
  }
}
