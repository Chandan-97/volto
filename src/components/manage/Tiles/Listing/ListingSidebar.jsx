import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { injectIntl, intlShape } from 'react-intl';

@injectIntl
/**
 * Listing sidebar class
 * @class ListingSidebar
 * @extends Component
 */
export default class ListingSidebar extends Component {
  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    data: PropTypes.objectOf(PropTypes.any).isRequired,
    tile: PropTypes.string.isRequired,
    intl: intlShape.isRequired,
    onChangeTile: PropTypes.func.isRequired,
  };

  /**
   * Constructor
   * @method constructor
   * @param {object} props Properties
   * @returns {undefined}
   */
  constructor(props) {
    super(props);
    this.state = {
      query: props.data.query || '',
    };
  }

  /**
   * Change handler
   * @method onChange
   * @param {object} e Change event
   * @returns {undefined}
   */
  onChange = e => {
    const value = e.currentTarget.value;
    this.setState({
      query: value,
    });
  };

  /**
   * Submit handler
   * @method onSubmit
   * @param {object} e Submit event
   * @returns {undefined}
   */
  onSubmit = e => {
    e.preventDefault();
    const { data, onChangeTile, tile } = this.props;
    const { query } = this.state;
    onChangeTile(tile, {
      ...data,
      query,
    });
  };

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    return (
      <div className="listing-sidebar">
        <form className="query-form" onSubmit={this.onSubmit}>
          <input
            name="query"
            className="query"
            placeholder="querystring"
            onChange={this.onChange}
            value={this.state.query}
          />
        </form>
      </div>
    );
  }
}
