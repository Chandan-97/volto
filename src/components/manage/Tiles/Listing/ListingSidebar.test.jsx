import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';

import ListingSidebar from './ListingSidebar';

const mockStore = configureStore();
const baseStore = {
  intl: {
    locale: 'en',
    messages: {},
  },
};

test('renders the listing tile sidebar', () => {
  const store = mockStore(baseStore);
  const component = renderer.create(
    <Provider store={store}>
      <ListingSidebar data={{}} tile="test" onChangeTile={() => {}} />
    </Provider>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
