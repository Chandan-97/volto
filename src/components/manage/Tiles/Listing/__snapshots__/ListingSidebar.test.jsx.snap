// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the listing tile sidebar 1`] = `
<div
  className="listing-sidebar"
>
  <form
    className="query-form"
    onSubmit={[Function]}
  >
    <input
      className="query"
      name="query"
      onChange={[Function]}
      placeholder="querystring"
      value=""
    />
  </form>
</div>
`;
